packages/compass/src/main/auto-update-manager.ts

```diff
@@ -827,11 +827,53 @@ class CompassAutoUpdateManager {
     this.setState(AutoUpdateManagerState.RestartDismissed);
   }
 
-  private static _init(
+  private static checkForMismatchedMacOSArch() {
+    const mismatchedOnArm =
+      isMismatchedArchDarwin() && getSystemArch() === 'arm64';
+
+    if (!mismatchedOnArm) {
+      return;
+    }
+
+    void dialog
+      .showMessageBox({
+        icon: COMPASS_ICON,
+        message: 'Mismatched architecture detected',
+        detail:
+          'You are currently using a build of Compass that is not optimized for Apple Silicon processors. This version might have significat performance issues when used. ' +
+          'Would you like to download the version of Compass optimized for Apple Silicon processors now?',
+        buttons: [
+          'Download Compass for Apple Silicon (Recommended)',
+          'Not now',
+        ],
+        cancelId: 1,
+      })
+      .then(({ response }) => {
+        if (response === 0) {
+          return dl.download(
+            BrowserWindow.getAllWindows()[0],
+            `https://compass.mongodb.com/api/v2/download/${this.autoUpdateOptions.version}/compass/${this.autoUpdateOptions.channel}/darwin-arm64`
+          );
```

Comment:
I'll change this to `shell.openExternal` I think, it's so slow that the in-app download is misbehaving